unity/CITools/BuildDriver/Program.cs

```diff
@@ -1,110 +1,172 @@
-﻿using System.Runtime.InteropServices;
+﻿using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Runtime.InteropServices;
 using NiceIO;
 
 namespace BuildDriver;
 
 public class Program
 {
     public static bool RunningOnYamato() => Environment.GetEnvironmentVariable("YAMATO_PROJECT_ID") != null;
-    public static async Task<int> Main(string[] args)
+    public static int Main(string[] args)
     {
-        string architecture = GetArchitectures()[0]; // Get default arch for current system
-        string configuration = "Release";
-        bool silent = false;
-        bool skipBuild = false;
-        TestTargets testTargets = TestTargets.None;
-        bool zip = RunningOnYamato();
-        Task<NPath>? zipTask = null;
-        foreach (string arg in args)
+        var buildOption = new Option<BuildTargets>("--build", "--b")
         {
-            if ((arg.StartsWith("--arch") || arg.StartsWith("--architecture="))
-                && !TryParseArgument(GetArchitectures(true), arg, out architecture))
-                return 1;
-            else if ((arg.StartsWith("--config=") || arg.StartsWith("--configuration="))
-                     && !TryParseArgument(new[] {"Release", "Debug"}, arg, out configuration))
-                return 1;
-            else if (arg.Equals("--silent") || arg.Equals("--s"))
-                silent = true;
-            else if (arg.Equals("--zip") || arg.Equals("--z"))
-                zip = true;
-            else if (arg.Equals("--skip-build"))
-                skipBuild = true;
-            else if (arg.Equals("--test"))
+            AllowMultipleArgumentsPerToken = true,
+            Arity = ArgumentArity.ZeroOrMore,
+            CustomParser = result =>
             {
-                skipBuild = true; // Assume we've already built
-                testTargets = TestTargets.All;
-            }
-            else if (arg.StartsWith("--test="))
+                if (result.Tokens.Count == 0)
+                    return BuildTargets.All; //For when --build is provided without any arguments.
```

Comment:
@joncham Just added these for `--build` and `--test`. Was doing some testing and saw that the tests weren't running when `--test` was provided without any arguments which didn't make sense.